json_schema/core/file/file_core.json

```
@@ -49,6 +49,17 @@
             "type": "string",
             "user_friendly": "Checksum",
             "example": "e09a986c2e630130b1849d4bf9a94c06"
-	    }
+	    },
+        "file_source": {
+            "description": "Has the file come from a contributor or the DCP?",
```

(optional) Alternat description: The name of the entity that generated the file

---

I think that may be a little confusing to some users (e.g. people may understand that field as the institution they are working in), but happy to discuss about it

---

The term entity is already taken.

The phrasing is a bit awkward but more importantly, I'd avoid repeating the enum values in the description. We'd have to rewrite it every time a new file source needs to be added.

```suggestion
            "description": "The source of the file. This is typically an organisation, person or dedicated process.",
```